datalad/cmd.py

```diff
@@ -353,6 +353,111 @@ def log(self, msg, level=logging.DEBUG):
             lgr.log(level, "{%s} %s" % (self.protocol.__class__.__name__, msg))
 
 
+class GitRunner(Runner):
+    """
+    Overloads the runner class to check update GIT_DIR and GIT_WORK_TREE environment variables set to the correct path
+    """
+
+    def run(self, cmd, log_stdout=True, log_stderr=True, log_online=False,
+            expect_stderr=False, expect_fail=False,
+            cwd=None, env=None, shell=None):
+
+        # assert_true(cwd, None)
+        # assert_false(self.cwd, None)
+        env = {"GIT_DIR": self.cwd+"/.git", "GIT_WORK_TREE": self.cwd+"/"}  # set environment variables, rm chdir error
+
+        cwd = os.getcwd()   # store current working directory
+        # lgr.error("INFO: set environment to %s based on git_root @ %s, cwd @ %s" % (env, self.cwd, os.getcwd()))
+
+        # chpwd(self.cwd)     # move to git-root
+        # lgr.error("INFO: moved to git_root @ %s" % (os.getcwd()))
+
+        outputstream = subprocess.PIPE if log_stdout else sys.stdout
```

Comment:
all below is not necessary. You should just call superclass's run: `return super(GitRunner, self).run(...env=adjusted_env)`